crates/transport/src/dual.rs

```diff
@@ -0,0 +1,191 @@
+use crate::{TransportError, TransportFut};
+use alloy_json_rpc::{RequestPacket, ResponsePacket};
+use tower::Service;
+
+/// Trait for converting two transports into a dual transport.
+pub trait DualTransportHandler<L, R> {
+    /// The type of the future returned by the transport.
+    fn call(&self, request: RequestPacket, left: L, right: R) -> TransportFut<'static>;
+}
+
+impl<F, L, R> DualTransportHandler<L, R> for F
+where
+    F: Fn(RequestPacket, L, R) -> TransportFut<'static> + Send + Sync,
+    L: Service<RequestPacket, Future = TransportFut<'static>, Error = TransportError> + Clone,
+    R: Service<RequestPacket, Future = TransportFut<'static>, Error = TransportError> + Clone,
```

Comment:
I think we likely don't need these bounds here, because those will be enforced by the the transport